lib/util/downloadGoogleFonts.js

```diff
@@ -0,0 +1,120 @@
+ * User agent strings by desired font format.
+ * @readonly
+ * @enum {string}
+ */
+const formatAgents = {
+  eot:
+    'Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1; WOW64; Trident/4.0; SLCC2; .NET CLR 2.0.50727; .NET CLR 3.5.30729; .NET CLR 3.0.30729; .NET4.0C; .NET4.0E)',
+  ttf:
+    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_6_8) AppleWebKit/534.59.8 (KHTML, like Gecko) Version/5.1.9 Safari/534.59.8',
+  woff:
+    'Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; .NET CLR 2.0.50727; .NET CLR 3.0.30729; .NET CLR 3.5.30729; rv:11.0) like Gecko',
+  woff2:
+    'Mozilla/5.0 (Windows NT 10.0; Win64; x64; ServiceUI 8) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393'
+};
+
+/**
+ * Download google fonts for self-hosting
+ *
+ * @async
+ * @param  {FontProps[]} fontProps List of CSS font properties to get fonts for
+ * @param  {Object} options
+ * @param  {String[]} [options.formats=['woff2', 'woff']] List of formats that should be inclued in the output
+ * @param  {String} [options.fontDisplay='swap'] CSS font-display value in returned CSS blocks
+ * @param  {String} [options.text] Text to create a subset with
+ * @return {String[]} CSS assets with inlined google fonts
+ */
+async function downloadGoogleFonts(
+  fontProps,
+  { formats = ['woff2', 'woff'], fontDisplay = 'swap', text } = {}
+) {
+  const sortedFormats = [];
+
+  for (const format of formatOrder) {
+    if (formats.includes(format)) {
+      sortedFormats.push(format);
+    }
+  }
+
+  const results = await Promise.all(
+    fontProps.map(async originalProps => {
+      const fontProps = {};
+      const googleFontId = getGoogleIdForFontProps(originalProps);
+      let fontCssUrl = `https://fonts.googleapis.com/css?family=${googleFontId}`;
+
+      if (text) {
+        fontCssUrl += `&text=${encodeURIComponent(text)}`;
+      }
+
+      fontProps.src = await Promise.all(
+        sortedFormats.map(async format => {
+          const assetGraph = new AssetGraph();
+          assetGraph.teepee.headers['User-Agent'] = formatAgents[format];
+
+          const [cssAsset] = await assetGraph.loadAssets(fontCssUrl);
+
+          await assetGraph.populate();
+
+          const [fontRelation] = assetGraph.findRelations({
+            from: cssAsset,
+            type: 'CssFontFaceSrc'
+          });
+
+          fontRelation.node.each(decl => {
+            if (decl.prop !== 'src') {
+              fontProps[decl.prop] = decl.value;
+            }
+          });
+
+          return [fontRelation.to, fontRelation.format];
+        })
+      );
+
+      if (!('unicode-range' in fontProps)) {
+        const font = fontProps.src[0][0];
+        fontProps['unicode-range'] = unicodeRange(
+          fontkit.create(font.rawSrc).characterSet
+        );
+      }
+
+      return fontProps;
+    })
+  );
+
+  return results.map(result => {
+    const declarationStrings = [];
+
+    Object.entries(result).map(([property, value]) => {
```

Comment:
Could you use a `for ... of` here? It's a bit weird to use `Array#map` and discard the return value.